docs/ref/models/fields.txt

```diff
@@ -1116,7 +1116,7 @@ Django also defines a set of fields that represent relations.
 ``ForeignKey``
 --------------
 
-.. class:: ForeignKey(othermodel, [**options])
+.. class:: ForeignKey(othermodel, on_delete, [**options])
```

Comment:
@fcurella Yes, exactly!

Only thing is that since you need two deprecation shims regardless, I would instead put them in `ForeignKey` and `OneToOneField`, the two public API classes. You can modify the signature of `ForeignObject` freely without concern for backwards compatibility, because it's undocumented internal API. This will allow you to keep a more sensible signature for `ForeignObject` (with `to_fields` and `from_fields` still adjacent, and `from_fields` still required).